keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -68,7 +72,39 @@ class DeliveryConfigController(
       .let { processedDeliveryConfig ->
         validator.validate(processedDeliveryConfig)
         log.debug("Upserting delivery config '${processedDeliveryConfig.name}' for app '${processedDeliveryConfig.application}'")
-        repository.upsertDeliveryConfig(processedDeliveryConfig)
+        val config = repository.upsertDeliveryConfig(processedDeliveryConfig)
+        publisher.publishEvent(DeliveryConfigChangedNotification(config))
+        return config
+      }
+
```

Comment:
here is the new endpoint that I'm experimenting with. It takes both a delivery config and also git metadata which I will scrape from the git trigger of pipelines (if it exists)

Reply:
Update - I'm going to move the gitMetadata to the "metadata" section of the delivery config like @luispollo suggested, and then strip it so we don't save it to the database.